src/arviz_plots/backend/plotly/legend.py

```diff
@@ -36,25 +63,84 @@ def legend(
     None
         The legend is added to the target figure inplace
     """
-    if artist_kwargs is None:
-        artist_kwargs = {}
+    figure = plot_collection.get_viz("figure")
+    if "legend" in plot_collection.viz.children:
+        legend_number = len(plot_collection.viz["legend"].data_vars) + 1
+        legend_id = f"legend{legend_number}"
+    else:
+        legend_number = 1
+        legend_id = "legend"
+    kwargs.setdefault("legend_y", {1: 1, 2: 0, 3: 0.5}[legend_number])
+    kwargs["legend_title_text"] = title
+    legend_kwargs = kwargs.pop(legend_id, {}).copy()
+    kwargs_list = list(kwargs.items())
+    for key, value in kwargs_list:
+        if key.startswith("legend"):
+            kwargs.pop(key)
+            legend_kwargs[key[len("legend_") :]] = value
+    kwargs[legend_id] = legend_kwargs
+    if visual_kwargs is None:
+        visual_kwargs = {}
+    else:
+        visual_kwargs = visual_kwargs.copy()
 
-    if artist_type == "line":
-        artist_fun = target.add_scatter
+    if visual_type == "line":
+        visual_fun = figure.add_scatter
         kwarg_list = [dealiase_line_kwargs(**kws) for kws in kwarg_list]
         mode = "lines"
+        visual_kwargs.setdefault("line_color", "black")
     else:
         raise NotImplementedError("Only line type legends supported for now")
 
+    if update_visuals:
+        for group, viz_data in plot_collection.viz.children.items():
+            if group in {"plot", "row_index", "col_index"}:
+                continue
+            viz_ds = viz_data.dataset
+            if any((d not in viz_ds.dims) and (d != "__variable__") for d in legend_dim):
+                continue
+            for var_name, sel, _ in xarray_sel_iter(viz_ds, skip_dims={}):
+                target_viz = viz_ds[var_name].sel(sel).item()
+                if target_viz is None:
+                    continue
+                if not isinstance(target_viz, (Scatter, Bar)):
+                    break
+
+                def trace_matcher(trace):
+                    return (
+                        (getattr(trace, "mode", "na") == getattr(target_viz, "mode", "na"))
+                        and (getattr(trace, "line", "na") == getattr(target_viz, "line", "na"))
+                        and (trace.marker == target_viz.marker)
+                        and (trace.text == target_viz.text)
+                        and (trace.x.shape == target_viz.x.shape)
+                        and np.allclose(trace.y, target_viz.y)
+                        and np.allclose(trace.x, target_viz.x)
+                    )
+
+                target_plot = plot_collection.get_target(var_name, sel)
```

Comment:
Is this `.get_target(var_name, sel)` similar to `viz_ds[var_name].sel(sel).item()`?

I imagine they're not exactly the same, but close.
It would be good to have a comment explaining what the bigger loop does. 
Something along the lines of "Updates the legend group for visuals in the plot that share this and that, only when X and Y"